account_invoice_secondary_unit/models/sale_order_line.py

```diff
@@ -12,6 +13,16 @@ def _prepare_invoice_line(self, qty):
         self.ensure_one()
         res = super(SaleOrderLine, self)._prepare_invoice_line(qty)
         if self.secondary_uom_id:
-            res['secondary_uom_qty'] = self.secondary_uom_qty
+            res['secondary_uom_qty'] = self._get_secondary_uom_qty()
             res['secondary_uom_id'] = self.secondary_uom_id.id
         return res
+
+    @api.multi
+    def _get_secondary_uom_qty(self):
+        factor = self.secondary_uom_id.factor * \
+            self.product_uom.factor
+        rounding_factor = self.secondary_uom_id.uom_id.factor
+        return float_round(
+            self.qty_delivered / (factor or 1.0),
```

Comment:
We should use `qty` instead of `self.qty_delivered` here.

Ref: https://github.com/odoo/odoo/blob/389bd53a0bcde0d5911f15a09cd03a84678b5380/addons/sale/models/sale.py#L1373